sklearn/manifold/isomap.py

```diff
@@ -100,19 +102,26 @@ def __init__(self, n_neighbors=5, n_components=2, eigen_solver='auto',
         self.n_jobs = n_jobs
 
     def _fit_transform(self, X):
-        X = check_array(X)
+
+        if self.neighbors_algorithm == 'precomputed':
+            algorithm = 'brute'
```

Comment:
codecov says this is never run. You don't appear to have a test_isomap.py change

And is there a reason we do not support a `metric` param here?

Reply:
> And is there a reason we do not support a metric param here?

I don't know